### PR TITLE
PT-8074: Modules healthcheck

### PR DESCRIPTION
**Pull request by getmansky**
## Description
Modules and cache healthcheck
### Jira-link: https://virtocommerce.atlassian.net/browse/PT-8074
https://virtocommerce.atlassian.net/browse/PT-8080

### CODE REVIEW

**Review by ksavosteev — APPROVED** [review]

File: src/VirtoCommerce.Platform.Web/Infrastructure/HealthCheck/CacheHealthChecker.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        public CacheHealthChecker(IPlatformMemoryCache memoryCache)
+        {
+            _memoryCache = memoryCache;
+        }
+
+        public Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            if ((_memoryCache as PlatformMemoryCache)?.CacheEnabled ?? false)
+            {
+                return Task.FromResult(HealthCheckResult.Degraded("Cache is off"));
```
Comment (Vectorfield4) [inline src/VirtoCommerce.Platform.Web/Infrastructure/HealthCheck/CacheHealthChecker.cs:22]: Use unhealthy.

File: src/VirtoCommerce.Platform.Web/Infrastructure/HealthCheck/ModulesHealthChecker.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                    .ToDictionary(
+                        x => x.Id,
+                        x => new ModulesHealthReportRecord
+                        {
+                            Title = x.Title,
+                            Version = x.Version.ToString(),
+                            Errors = x.Errors.ToArray()
+                        } as object);
+
+                return Task.FromResult(HealthCheckResult.Degraded("Some modules have errors", null, errorsDictionary));
```
Comment (Vectorfield4) [inline src/VirtoCommerce.Platform.Web/Infrastructure/HealthCheck/ModulesHealthChecker.cs:37]: Please, add label for `null`.

File: src/VirtoCommerce.Platform.Web/Infrastructure/HealthCheck/ModulesHealthChecker.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        {
+            _localModuleCatalog = localModuleCatalog;
+        }
+
+        public Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            var errorModules = _localModuleCatalog.Modules
+                .OfType<ManifestModuleInfo>()
+                .Where(x => x.Errors.Any())
+                .ToArray();
```
Comment (Vectorfield4) [inline src/VirtoCommerce.Platform.Web/Infrastructure/HealthCheck/ModulesHealthChecker.cs:23]: Use `ToDictionary()` here. No reason to sequence serialization.

File: src/VirtoCommerce.Platform.Web/Infrastructure/HealthCheck/CacheHealthChecker.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using VirtoCommerce.Platform.Caching;
+using VirtoCommerce.Platform.Core.Caching;
+
+namespace VirtoCommerce.Platform.Web.Infrastructure.HealthCheck
+{
+    public class CacheHealthChecker : IHealthCheck
```
Comment (Vectorfield4) [inline src/VirtoCommerce.Platform.Web/Infrastructure/HealthCheck/CacheHealthChecker.cs:9]: add `sealed`

File: src/VirtoCommerce.Platform.Web/Infrastructure/HealthCheck/ModulesHealthChecker.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using VirtoCommerce.Platform.Core.Modularity;
+
+namespace VirtoCommerce.Platform.Web.Infrastructure.HealthCheck
+{
+    public class ModulesHealthChecker : IHealthCheck
```
Comment (Vectorfield4) [inline src/VirtoCommerce.Platform.Web/Infrastructure/HealthCheck/ModulesHealthChecker.cs:9]: add `sealed`

File: src/VirtoCommerce.Platform.Web/Startup.cs [hunk 6]
```diff
@@ -411,7 +415,13 @@ public void ConfigureServices(IServiceCollection services)
             // extra AI module and causes the hight CPU utilization and telemetry data flood on production env.
             services.AddAppInsightsTelemetry(Configuration);
 
-            services.AddHealthChecks();
+            services.AddHealthChecks()
+                .AddCheck<ModulesHealthChecker>("Modules health",
+                    failureStatus: HealthStatus.Degraded,
+                    tags: new[] { "Modules" })
+                .AddCheck<CacheHealthChecker>("Cache health",
+                    failureStatus: HealthStatus.Degraded,
```
Comment (Vectorfield4) [inline src/VirtoCommerce.Platform.Web/Startup.cs:423]:
```suggestion
                    failureStatus: HealthStatus.Unhealthy,
```

**Review by Vectorfield4 — CHANGES_REQUESTED** [review]

**Pull request merged**